README.md

```diff
@@ -149,67 +147,61 @@ final controller = CodeController(
 );
 ```
 
-This creates a section named `section1`.
-The built-in `BracketsStartEndNamedSectionParser` class is designed to parse sections
-from the code comments using the above syntax.
-It also hides any single-line comment that has a section label with the above syntax,
-although such comments are still present in the editor's hidden state
-and will be revealed when copying the text.
+The example above creates a section named `section1`.
+The built-in `BracketsStartEndNamedSectionParser` class is designed to parse sections from the code comments using the above syntax. It also hides any single-line comment that has a section label with the above syntax,
+although such comments are still present in the editor's hidden state and will be revealed when copying the text.
+
+To customize parsing sections using any other syntax, subclass `AbstractNamedSectionParser`.
 
-If you need to parse sections using any other syntax, subclass `AbstractNamedSectionParser`.
+## Read-Only Code Blocks
 
-## Partial Locking
+Flutter Code Editor allows to define read-only code blocks, that might be useful for learning use cases when users are guided to modify certain code blocks, and other code desired to be protected from modification. 
 
-This is how you make parts of the code read-only:
+To make a `named section` read-only, pass set of `named sections` to the `controller.readOnlySectionNames`:
 
 ```dart
 controller.readOnlySectionNames = {'section1'};
 ```
 
-This locks the given sections. Any non-existent section names in this set are ignored.
-To make the code editable again, pass an updated set.
+This locks the given sections from modifications in the Flutter Code Editor. Any non-existent section names in this set are ignored. To make the code editable again, pass an updated set to `controller.readOnlySectionNames`.
 
-When using this feature, `text` and `value` properties cannot be used to change the text
-programmatically because they have the same effect as the user input,
-and so locking affects them as well.
+When using this feature, `text` and `value` properties cannot be used to change the text programmatically because they have the same effect as the user input, and so locking affects them as well.
```

Comment:
Single line breaks are ignored in markdown so I use them to keep lines short. If possible, please do not delete them. This also minimizes false-positive diffs.